x-pack/qa/core-rest-tests-with-security/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms.yml

```diff
@@ -0,0 +1,109 @@
+        username: "non-synonyms-user"
+        body:
+          password: "non-synonyms-user-password"
+          roles : [ "non-synonyms-role" ]
+
+  - do:
+      security.put_role:
+        name: "non-synonyms-role"
+        body:
+          indices:
+            - names: ["*"]
+              privileges: [ "manage", "write", "read" ]
+
+---
+"Create synonyms set - no manage_search_synonyms privilege":
+  - do:
+      catch: "forbidden"
+      headers: { Authorization: "Basic bm9uLXN5bm9ueW1zLXVzZXI6bm9uLXN5bm9ueW1zLXVzZXItcGFzc3dvcmQ=" }  # non-synonyms-user
+      synonyms.put:
+        synonyms_set: test-synonyms
+        body:
+          synonyms_set:
+            - synonyms: "hello, hi"
+              id: "test-id"
+
+---
+"Get synonyms set - no manage_search_synonyms privilege":
+  - do:
+      catch: "forbidden"
+      headers: { Authorization: "Basic bm9uLXN5bm9ueW1zLXVzZXI6bm9uLXN5bm9ueW1zLXVzZXItcGFzc3dvcmQ=" }  # non-synonyms-user
+      synonyms.get:
+        synonyms_set: test-synonyms
+
+---
+"Delete synonyms set - no manage_search_synonyms privilege":
+  - do:
+      catch: "forbidden"
+      headers: { Authorization: "Basic bm9uLXN5bm9ueW1zLXVzZXI6bm9uLXN5bm9ueW1zLXVzZXItcGFzc3dvcmQ=" }  # non-synonyms-user
+      synonyms.delete:
+        synonyms_set: test-synonyms
+---
+"List synonyms sets - no manage_search_synonyms privilege":
+  - do:
+      catch: "forbidden"
+      headers: { Authorization: "Basic bm9uLXN5bm9ueW1zLXVzZXI6bm9uLXN5bm9ueW1zLXVzZXItcGFzc3dvcmQ=" }  # non-synonyms-user
+      synonyms_sets.get: { }
+
+---
+"Update a synonyms rule - no manage_search_synonyms privilege":
+  - do:
+      catch: "forbidden"
+      headers: { Authorization: "Basic bm9uLXN5bm9ueW1zLXVzZXI6bm9uLXN5bm9ueW1zLXVzZXItcGFzc3dvcmQ=" }  # non-synonyms-user
+      synonym_rule.put:
+        synonyms_set: "test-synonyms"
+        synonym_rule: "test-id-2"
+        body:
+          synonyms: "bye, goodbye, seeya"
+
+---
+"Get a synonym rule - no manage_search_synonyms privilege":
+  - do:
+      catch: "forbidden"
+      headers: { Authorization: "Basic bm9uLXN5bm9ueW1zLXVzZXI6bm9uLXN5bm9ueW1zLXVzZXItcGFzc3dvcmQ=" }  # non-synonyms-user
+      synonym_rule.get:
+        synonyms_set: "test-synonyms"
+        synonym_rule: "test-id-2"
+
+---
+"Delete synonym rule - no manage_search_synonyms privilege":
+  - do:
+      catch: "forbidden"
+      headers: { Authorization: "Basic bm9uLXN5bm9ueW1zLXVzZXI6bm9uLXN5bm9ueW1zLXVzZXItcGFzc3dvcmQ=" }  # non-synonyms-user
+      synonym_rule.delete:
+        synonyms_set: test-synonyms
+        synonym_rule: test-id-2
+
+---
+"Create synonyms set - with manage_search_synonyms privilege":
+  - do:
+      headers: { Authorization: "Basic c3lub255bXMtdXNlcjpzeW5vbnltcy11c2VyLXBhc3N3b3Jk" }  # synonyms-user
+      synonyms.put:
+        synonyms_set: test-synonyms
+        body:
+          synonyms_set:
+            - synonyms: "hello, hi"
+              id: "test-id"
+
+  - match: { result: "created" }
```

Comment:
I think we should have more tests for the `synonyms-user` to show it can excercise most (if not all) actions similar to how `non-synonyms-user` is tested?

Reply:
Makes sense. I've split into two test files, one for testing unprivileged access and another for privileged access. LMKWYT!